plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -111,7 +111,10 @@ export class EventPipelineRunner {
     }
 
     async runEventPipelineSteps(event: PluginEvent): Promise<EventPipelineResult> {
-        if (this.hub.poeEmbraceJoinForTeams?.(event.team_id)) {
+        if (
+            this.hub.poeEmbraceJoinForTeams?.(event.team_id) ||
+            (event.team_id <= this.hub.POE_WRITES_ENABLED_MAX_TEAM_ID && !this.hub.poeWritesExcludeTeams(event.team_id))
+        ) {
```

Comment:
I'm a little confused why `poeEmbraceJoinForTeams` uses optional chaining here since `Hub` doesn't have it typed as an optional member. Is there some gotcha here that I'm not aware of that I should be carrying over to `poeWritesExcludeTeams` as well?

Reply:
should be fine to not copy it, the type safety should indeed be enough here